doc/api/http2.md

```diff
@@ -2379,6 +2379,9 @@ changes:
     serialized, compressed block of headers. Attempts to send headers that
     exceed this limit will result in a `'frameError'` event being emitted
     and the stream being closed and destroyed.
+    While this sets the maximum allowed size to the entire block of headers,
+    `nghttp2` (the internal http2 library) has a limit of `65536`
+    for each key/value pair.
```

Comment:
Is that the compressed or decompressed maximum? (Decompressed, I think.)

Looking at nghttp2_hd.c, that limit seems to only apply when receiving headers? The previous paragraph says "[..] Attempts to send headers [..]".

Reply:
> Is that the compressed or decompressed maximum? (Decompressed, I think.)

Exactly

> Looking at nghttp2_hd.c, that limit seems to only apply when receiving headers? The previous paragraph says "[..] Attempts to send headers [..]".

Actually, it refers to both sides, if you try to send a `trailers` bigger than nghttp2 threshold, it will throw an ERR_FRAME_SIZE_ERROR (https://github.com/nghttp2/nghttp2/blob/master/lib/nghttp2_session.c#L2155) as expected by RFC7540. See https://github.com/nodejs/node/pull/42147.

I thought in adding this note to `clientsession.request({ headers })` documentation as well, but, once it is a threshold to the server itself, doesn't make sense to add this note there, and currently we don't have a property for the header size received.

What do you think?

Reply:
Maybe explicitly call out it's the decompressed limit but that's just a suggestion.